gobblin-runtime/src/main/java/gobblin/runtime/listeners/JobListeners.java

```diff
@@ -73,13 +74,55 @@ public Void call()
     }
 
     @Override
-    public void onJobCancellation(final JobState jobState) {
+    public void onJobStart(final JobContext jobContext) {
       for (final JobListener jobListener : jobListeners) {
         this.completionService.submit(new Callable<Void>() {
           @Override
           public Void call()
               throws Exception {
-            jobListener.onJobCancellation(jobState);
+            jobListener.onJobStart(jobContext);
+            return null;
+          }
+        });
+      }
+    }
+
+    @Override
+    public void onJobCompletion(final JobContext jobContext) {
+      for (final JobListener jobListener : jobListeners) {
+        this.completionService.submit(new Callable<Void>() {
+          @Override
+          public Void call()
+              throws Exception {
+            jobListener.onJobCompletion(jobContext);
```

Comment:
If any `jobListener` throws an exception, the remaining ones (if any) will be skipped. Is this behavior desirable?